src/components/product-switcher/index.tsx

```diff
@@ -0,0 +1,208 @@
+    product: Product,
+    index: number
+  ): ReactElement => {
+    const isCurrentProduct = product.slug === currentProduct.slug
+
+    const handleAnchorKeyDown: KeyboardEventHandler<HTMLAnchorElement> = (
+      e
+    ) => {
+      const lastIndex = products.length - 1
+      const isFirstItem = index === 0
+      const isLastItem = index === lastIndex
+      if (!(isFirstItem || isLastItem)) {
+        return
+      }
+
+      const isTabbingForward = !e.shiftKey && e.key === 'Tab'
+      const isTabbingBackward = e.shiftKey && e.key === 'Tab'
+      if (!(isTabbingForward || isTabbingBackward)) {
+        return
+      }
+
+      const anchorElements = getAllAnchorElements()
+      if (isFirstItem && isTabbingBackward) {
+        const lastAnchorElement = anchorElements[lastIndex]
+        lastAnchorElement.focus()
+        e.preventDefault()
+      } else if (isLastItem && isTabbingForward) {
+        const firstAnchorElement = anchorElements[0]
+        firstAnchorElement.focus()
+        e.preventDefault()
+      }
+    }
+
+    return (
+      <li
+        className={s.switcherOption}
+        id={generateSwitcherOptionIdFromProduct(product)}
+        key={product.slug}
+      >
+        <a
+          aria-current={isCurrentProduct ? 'page' : undefined}
+          className={classNames(
+            s.switcherOptionAnchor,
+            s.switcherOptionContainer
+          )}
+          href={product.url}
+          onKeyDown={handleAnchorKeyDown}
+        >
+          <ProductIcon product={product.slug} />
+          <span>{product.name}</span>
+        </a>
+      </li>
+    )
+  }
+
+  const renderProductGroup = (productGroup: ProductGroup, index: number) => {
+    const shouldRenderHorizontalRule = index > 0
+    return (
+      <ul key={`product-group-${index}`} role="group">
+        {shouldRenderHorizontalRule && <hr />}
+        {productGroup.map(renderProductListItem)}
+      </ul>
+    )
+  }
+
+  /**
+   * I _think_ we want the containing element to be a nav, currently clashes with other
+   * styles so not using that element just yet
+   */
+  return (
+    <div className={s.productSwitcher} ref={productChooserRef}>
+      <button
+        aria-controls={OPTION_LIST_ID}
+        aria-expanded={isOpen}
+        aria-labelledby={generateSwitcherOptionIdFromProduct(currentProduct)}
+        className={s.switcherButton}
+        onClick={() => {
+          setIsOpen(!isOpen)
+        }}
+        onKeyDown={handleKeyDown}
+        ref={buttonRef}
+      >
+        <div className={s.switcherOptionContainer}>
+          <ProductIcon product={currentProduct.slug} />
+          <span>{currentProduct.name}</span>
+        </div>
+        <IconCaret16 className={s.switcherCaret} />
+      </button>
+      <ul
+        className={s.switcherOptionList}
+        id={OPTION_LIST_ID}
+        onKeyDown={handleKeyDown}
+        style={{ display: isOpen ? 'block' : 'none' }}
+      >
+        {products.map(renderProductGroup)}
```

Comment:
Looks like we're missing wrapping `li` around each group? Nested `ul's contain default margin 
<img width="408" alt="CleanShot 2021-12-15 at 16 51 30@2x" src="https://user-images.githubusercontent.com/825855/146270672-fceb99b3-c127-495a-84b8-54cfb7273b67.png">

Reply:
Sorry for the confusion! This isn't ready for review yet anymore. I've converted it back to a draft. I'm currently working on converting this to dark mode and also to include horizontal rules.